src/dawn/SIR/SIR.proto

```diff
@@ -0,0 +1,608 @@
+  Expr expr = 1;          // Expression to return
+  SourceLocation loc = 2; // Source location
+}
+
+// @brief Declaration of a variable
+//
+// Synopsis
+// @code{.cpp}
+// Type name[dimension] op init_list
+// @endcode
+//
+// @b Example
+// In the following C++ expression:
+//
+// @code{.cpp}
+//  double foo[2] = {2.3, 5.3};
+// @endcode
+//
+// the members of `VarDeclStmt` would be:
+//
+// @code{.cpp}
+//  - type: Float
+//  - name: "foo"
+//  - dimension: 2
+//  - op: "="
+//  - init_list: {LiteralAccessExpr{"2.3"}, LiteralAccessExpr{"5.3"}}
+// @endcode
+//
+// @ingroup sir_proto
+message VarDeclStmt {
+  Type type = 1;       // Type of the variable
+  string name = 2;     // Name of the variable
+  int32 dimension = 3; // Dimension of the array or 0 for variables
+  string op = 4;       // Operation used for initialization
+
+  // List of expression used for array initialization or just 1 element for variable initialization
+  repeated Expr init_list = 5;
+
+  SourceLocation loc = 6; // Source location
+}
+
+// @brief Declaration of a vertical region
+//
+// @ingroup sir_proto
+message VerticalRegionDeclStmt {
+  VerticalRegion vertical_region = 1;
+  SourceLocation loc = 2; // Source location
+}
+
+// @brief Declaration of a stencil call
+//
+// @ingroup sir_proto
+message StencilCallDeclStmt {
+  StencilCall stencil_call = 1;
+  SourceLocation loc = 2; // Source location
+}
+
+// @brief Declaration of a boundary condition
+//
+// @ingroup sir_proto
+message BoundaryConditionDeclStmt {
+  string functor = 1;        // Identifier of the boundary condition functor
+  repeated Field fields = 2; // List of field arguments to apply the functor to
+  SourceLocation loc = 3;    // Source location
+}
+
+// @brief Represent an if/then/else block
+//
+// Synopsis
+// @code{.cpp}
+//  if(cond_part)
+//    then_part
+//  else
+//    else_part
+// @endcode
+//
+// @ingroup sir_proto
+message IfStmt {
+  Stmt cond_part = 1;     // Condition (needs to be an `ExprStmt`)
+  Stmt then_part = 2;     // Then part
+  Stmt else_part = 3;     // Else part (can be empty)
+  SourceLocation loc = 4; // Source location
+}
+
+// @brief Apply the unary operations to an expression
+//
+// Synopsis
+// @code{.cpp}
+//  op operand
+// @endcode
+// @ingroup sir_proto
+message UnaryOperator {
+  string op = 1;          // Operation (e.g "+" or "-")
+  Expr operand = 2;       // Expression to apply the operation
+  SourceLocation loc = 3; // Source location
+}
+
+// @brief Apply a binary operation to `left` and `right`
+//
+// Synopsis
+// @code{.cpp}
+//  left op right
+// @endcode
+// @ingroup sir_proto
+message BinaryOperator {
+  Expr left = 1;          // Left-hand side
+  string op = 2;          // Operation (e.g "+" or "-")
+  Expr right = 3;         // Right-hand side
+  SourceLocation loc = 4; // Source location
+}
+
+// @brief Assignment operator
+//
+// Synopsis
+// @code{.cpp}
+//  left = right
+// @endcode
+// @ingroup sir_proto
+message AssignmentExpr {
```

Comment:
Assignment should support an operator as is currently in the c++ code to be able to capture assign and operation operators

Reply:
Fixed